appnav/src/main/kotlin/io/element/android/appnav/room/RoomFlowNode.kt

```diff
@@ -161,6 +161,14 @@ class RoomFlowNode @AssistedInject constructor(
                 }
             }
         }.launchIn(lifecycleScope)
+
+        // If the user leaves the room from this client, close the room flow.
+        membershipObserver.updates
+            .filter { it.roomId == roomId && !it.isUserInRoom }
+            .onEach {
```

Comment:
maybe add `.take(1)` to `navigateUp` only once?

Reply:
If we get two membership updates there is something wrong, but yeah sure, I can change